pkg/callback/scheduler_callback.go

```diff
@@ -118,15 +118,34 @@ func (callback *AsyncRMCallback) RecvUpdateResponse(response *si.UpdateResponse)
 	for _, release := range response.ReleasedAllocations {
 		log.Logger().Debug("callback: response to released allocations",
 			zap.String("UUID", release.UUID))
+		// delete the allocated pod
+		app := callback.context.GetApplication(release.ApplicartionID)
+		for _, task := range app.taskMap {
+			if task.allocationUUID == release.UUID {
+				// TerminationType 0 mean STOPPED_BY_RM
+				if release.TerminationType != si.AllocationRelease_STOPPED_BY_RM {
```

Comment:
we need to handle this check before iterating the task map in order to make this more efficient